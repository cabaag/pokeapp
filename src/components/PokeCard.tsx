import { Body, Card, CardItem, Text, Spinner, Row } from 'native-base';
import { Image, StyleSheet } from 'react-native';
import React, { memo } from 'react';
import { Pokemon } from '../types/Pokemon';

export interface PokeCardProps {
  pokemon: Pokemon
}

const styles = StyleSheet.create({
  sprite: {
    width: 100,
    height: 100
  },
  body: {
    flex: 1,
    alignContent: 'center',
    justifyContent: 'center',
    alignItems: 'center'
  }
})

const PokeCard: React.FC<PokeCardProps> = ({ pokemon }: PokeCardProps) => {
  return (
    <Card style={{ height: '100%', margin: 8 }}>
      <CardItem>
        <Body style={styles.body}>
          {
            pokemon?.sprites?.front_default ?
              <Image source={{ uri: pokemon.sprites.front_default }} style={styles.sprite} />
              : <Spinner />
          }
          <Text style={{ textTransform: 'capitalize' }}>
            {pokemon.name}
          </Text>

          <Row />

        </Body>
      </CardItem>
    </Card>
  );
}

export default memo(PokeCard);