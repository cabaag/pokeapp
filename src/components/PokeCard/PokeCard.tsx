import { Body, Card, Col, Row, Spinner, Text } from 'native-base';
import React, { memo } from 'react';
import { Image, StyleSheet, View } from 'react-native';
import typesMapper from 'utils/typesMapper';
import { Pokemon } from '../../types/Pokemon';

export interface PokeCardProps {
  pokemon: Pokemon;
  direction: 'column' | 'row';
}

const styles = StyleSheet.create({
  sprite: {
    width: 80,
    height: 80
  },
  body: {
    flex: 1,
    alignContent: 'center',
    justifyContent: 'center',
    alignItems: 'center'
  },
  badge: {
    borderRadius: 50,
    width: 10,
    height: 10,
    margin: 10,
  }
})

const PokeCard: React.FC<PokeCardProps> = ({ pokemon, direction }: PokeCardProps) => {
  const renderTypes = () => {
    return pokemon?.types?.map(type => (
      <View
        key={type.slot}
        style={{
          ...styles.badge,
          backgroundColor: typesMapper(type.type.name) as string
        }}
      />
    ))
  }
  return (
    <Card style={{ padding: 8 }}>
      <Body style={{ ...styles.body, flexDirection: direction }}>
        <Col>
          {direction === 'column' && (
            <Row style={{ justifyContent: 'center' }}>
              {renderTypes()}
            </Row>
          )}
          <Row style={{ justifyContent: 'center' }}>
            {
              pokemon?.sprites?.front_default ?
                <Image source={{ uri: pokemon.sprites.front_default }} style={styles.sprite} />
                : <Spinner />
            }
          </Row>
        </Col>
        <Col>
          {direction === 'row' && (
            <Row style={{ justifyContent: 'center' }}>
              {renderTypes()}
            </Row>
          )}
          <Row style={{ justifyContent: direction === 'row' ? 'flex-start' : 'center'}}>
            <Text style={{ fontWeight: 'bold' }}>
              #
              {' '}
              {pokemon?.id}
            </Text>
          </Row>
          <Row style={{ justifyContent: direction === 'row' ? 'flex-start' : 'center'}}>
            <Text style={{ textTransform: 'capitalize' }}>
              {pokemon.name}
            </Text>
          </Row>

        </Col>
      </Body>
    </Card>
  );
}

export default memo(PokeCard);