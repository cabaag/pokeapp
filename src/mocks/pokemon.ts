import { Pokemon } from '../types/Pokemon';

const PokemonMock: Pokemon = {
  base_experience: 0,
  height: 10,
  weight: 10,
  name: 'test',
  sprites: {
    front_default: '',
    front_shiny: '',
  },
  stats: [
    {
      base_stat: 10,
      effort: 0,
      stat: {
        name: 'stat test',
        url: '',
      },
    },
  ],
  id: 0,
};

export { PokemonMock };
