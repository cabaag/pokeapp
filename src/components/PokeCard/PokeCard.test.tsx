import { shallow, ShallowWrapper } from 'enzyme';
import { PokemonMock } from 'mocks/pokemon';
import React from 'react';
import PokeCard, { PokeCardProps } from './PokeCard';

const defaultProps: PokeCardProps = {
  pokemon: PokemonMock
}

function setup(): ShallowWrapper {
  return shallow(
    <PokeCard {...defaultProps} />
  )
}

describe('<PokeCard />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = setup();
  })

  it('matchs snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});