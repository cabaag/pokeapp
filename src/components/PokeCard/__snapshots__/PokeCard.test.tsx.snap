// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokeCard /> matchs snapshot 1`] = `
<Styled(Card)
  style={
    Object {
      "padding": 8,
    }
  }
>
  <Styled(Body)
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flex": 1,
        "flexDirection": undefined,
        "justifyContent": "center",
      }
    }
  >
    <ColumnNB>
      <RowNB
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <Styled(Spinner) />
      </RowNB>
    </ColumnNB>
    <ColumnNB>
      <RowNB
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          #
           
          0
        </Styled(Text)>
      </RowNB>
      <RowNB
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          test
        </Styled(Text)>
      </RowNB>
    </ColumnNB>
  </Styled(Body)>
</Styled(Card)>
`;
